Formula/asdf.rb

```diff
@@ -4,30 +4,40 @@ class Asdf < Formula
   url "https://github.com/asdf-vm/asdf/archive/v0.8.0.tar.gz"
   sha256 "9b667ca135c194f38d823c62cc0dc3dbe00d7a9f60caa0c06ecb3047944eadfa"
   license "MIT"
-  revision 1
+  revision 2
   head "https://github.com/asdf-vm/asdf.git"
 
   bottle :unneeded
-
-  depends_on "autoconf"
-  depends_on "automake"
   depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
-  depends_on "openssl@1.1"
-  depends_on "readline"
-  depends_on "unixodbc"
-
-  conflicts_with "homeshick",
-    because: "asdf and homeshick both install files in lib/commands"
+  depends_on "git"
 
   def install
     bash_completion.install "completions/asdf.bash"
     fish_completion.install "completions/asdf.fish"
     zsh_completion.install "completions/_asdf"
-    libexec.install "bin/private"
-    prefix.install Dir["*"]
-    touch prefix/"asdf_updates_disabled"
+    libexec.install Dir["*"]
+    touch libexec/"asdf_updates_disabled"
+    bin.write_exec_script (opt_libexec/"bin/asdf")
+    (prefix/"asdf.sh").write ". #{opt_libexec}/asdf.sh\n"
+    (prefix/"asdf.fish").write ". #{opt_libexec}/asdf.fish\n"
```

Comment:
@cho-m So I tested, and these two lines will make sure that currently documented (on asdf website) "caveat" works for sourcing. 

But I don't see how that prevents the need for a `rm -rf ~/.asdf/shims; asdf reshim` - can you confirm that old shims will point to the correct binary? 

My shims are defined as 

```shell
 cat .asdf/shims/iex    
#!/usr/bin/env bash
# asdf-plugin: elixir 1.11.3
exec /usr/local/opt/asdf/libexec/bin/asdf exec "iex" "$@"
```

The new executable asdf will no longer be versioned and instead always be the active one. I have opinions on this, but for now this will be the best approach until `$ asdf reshim` gets some fixes to it. 

But `/usr/local/opt/asdf/libexec/bin/asd` is not the same as the old path was. Which makes me think that the purposes of these two lines are negated since a migration step is necessary, at least for removing and reshimming.

Reply:
Using ASDF's documented installation steps with currently available Formula, my shims appear to be using `$HOMEBREW_PREFIX/opt/asdf/bin/asdf`. Pretty much multiple rounds of redirection.  Forcing a reshim can help remove one exec, but it at least works as is.

I'm on M1 mac so my exact path is different: `/opt/homebrew/opt/asdf/bin/asdf`.
Also, in my example, I'm using Unix/Linux-style XDG paths.

After installing patch:
```console
❯ brew info asdf | grep Cellar
/opt/homebrew/Cellar/asdf/0.8.0_2 (119 files, 244.0KB) *

❯ asdf current python
python          3.9.2 system    /Users/cho-m/.config/asdf/tool-versions

❯ python --version
Python 3.9.2

❯ pip --version
pip 21.0.1 from /Users/cho-m/.local/share/asdf/installs/python/3.9.2/lib/python3.9/site-packages/pip (python 3.9)

❯ python
Python 3.9.2 (default, Feb 21 2021, 22:26:49)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import toml
>>> toml.__file__
'/Users/cho-m/.local/share/asdf/installs/python/3.9.2/lib/python3.9/site-packages/toml/__init__.py'

❯ asdf current ruby
ruby            3.0.0           /Users/cho-m/.config/asdf/tool-versions

❯ ruby --version
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [arm64-darwin20]

❯ head -n3 "$ASDF_DATA_DIR/shims/ruby"
#!/usr/bin/env bash
# asdf-plugin: ruby 3.0.0
exec /opt/homebrew/opt/asdf/bin/asdf exec "ruby" "$@"
```

Reply:
@cho-m

Oh my God, this is amazing news.

@jthegedus This should be of interest to you. 

Just to see if I understood it...

Current shims are using `opt/asdf/bin/asdf`
My PR change adds a **new** `opt/asdf/bin/asdf` that points to `opt/asdf/libexec/bin/asdf` via
```ruby
    bin.write_exec_script (opt_libexec/"bin/asdf")
```
So new shims will get less indirection, but older shims still work just fine. 

I didn't actually expect that to migrate so well, this is amazing news. 
This combined with your added changes to create the files current users are currently sourcing 
```ruby
    (prefix/"asdf.sh").write ". #{opt_libexec}/asdf.sh\n"
```
Makes this a no change migration!
Correct? 



All that's left is discussing the caveat.

Reply:
Yes, that is what I see on my machine.
Should be a seamless migration for existing users.